### PR TITLE
Changed plot histogram code example and added another one in the API reference

### PR DESCRIPTION
**Pull request by Guillermo-Mijares-Vilarino**
### Summary
Changed the current `plot_histogram` code example in the [API reference](https://qiskit.org/documentation/stubs/qiskit.visualization.plot_histogram.html#qiskit.visualization.plot_histogram) and added another one.

### Details and comments
For the original code example I removed any references to the deprecated `execute` function and plotted two different executions in different colors instead of only one. 

The new example shows two different ways to order the bars using the `sort` argument: by descending probability order (`value_desc`) and by Hamming distance (`hamming`).  It also shows how one can easily represent two different histograms in the same code cell with the `display` function.

### CODE REVIEW

**Review by HuangJunye — COMMENTED** [review]
Thanks for working on this. I will modify your PR directly based on our discussions offline.

**Review by HuangJunye — APPROVED** [review]
LGTM! Thanks for making this contribution.

**Review by HuangJunye — APPROVED** [review]

File: qiskit/visualization/counts_visualization.py [hunk 1]
```diff
@@ -95,21 +95,61 @@ def plot_histogram(
...
+
+            transpiled_qc = transpile(qc, backend)
+            job = backend.run(transpiled_qc)
+            counts = job.result().get_counts()
+
+            # Sort by the probability in descendant order
+            hist1 = plot_histogram(counts, sort='value_desc')
+
+            # Sort by the hamming distance (the number of bit flips to change from
+            # one bitstring to the other)from a target string.
```
Comment (1ucian0) [inline qiskit/visualization/counts_visualization.py:148]:
```suggestion
            # one bitstring to the other) from a target string.
```

File: qiskit/visualization/counts_visualization.py [hunk 2]
```diff
@@ -95,21 +95,61 @@ def plot_histogram(
...
+            legend = ['First execution', 'Second execution']
+
+            plot_histogram([counts1, counts2], legend=legend, color=['crimson','midnightblue'], title="New Histogram")
+
+        .. jupyter-execute::
+
+            # You can sort the bitstrings using different methods.
+
+            from qiskit import transpile
+            import numpy as np
```
Comment (1ucian0) [inline qiskit/visualization/counts_visualization.py:129]:
```suggestion
            import numpy as np
            from qiskit import transpile
```

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
No sure.. this is getting into how-to land. The original example was much simpler.

Even more, in the same spirit as https://github.com/Qiskit/qiskit-terra/pull/8350, this could be as simple as
```python
plot_histogram({'00': 525, '11': 499})
```
I gave you this is a bit of the opposite extreme. I'm trying to make a point. An example is a stand-alone situation. With less other unrelated things.

**Comment by Guillermo-Mijares-Vilarino** [comment]
If you are talking about the second example @1ucian0, this was done so there were enough bars with varied height to show different ways to sort them. With 2 or 3 bars it's hard to observe.

**Comment by 1ucian0** [comment]
> If you are talking about the second example @1ucian0, this was done so there were enough bars with varied height to show different ways to sort them. With 2 or 3 bars it's hard to observe the difference.

```python
from qiskit.visualization import plot_histogram

counts = {'001': 596, '011': 211, '010': 50, '000': 117, '101': 33, '111': 8, '100': 6, '110': 3}
hist1 = plot_histogram(counts, sort='value_desc')
hist2 = plot_histogram(counts, sort='hamming', target_string='001')
display(hist1, hist2)
```

**Comment by HuangJunye** [comment]
@Guillermo-Mijares-Vilarino Can you please simplify all the examples to use counts dictionary instead of result.get_counts similar to the second example based on Luciano's comment? https://github.com/Qiskit/qiskit-terra/pull/8351#issuecomment-1206306744

**Comment by HuangJunye** [comment]
@Guillermo-Mijares-Vilarino You can take a look at the examples I created for the module level api page: https://github.com/Qiskit/qiskit-terra/pull/8569

**Comment by Guillermo-Mijares-Vilarino** [comment]
> @Guillermo-Mijares-Vilarino Can you please simplify all the examples to use counts dictionary instead of result.get_counts similar to the second example based on Luciano's comment? [#8351 (comment)](https://github.com/Qiskit/qiskit-terra/pull/8351#issuecomment-1206306744)

If I do that at least to the first example from this PR it would become practically identical to one of the examples you added in #8569 @HuangJunye. We should then decide which one to keep.

This is the only of my code examples PRs that uses any counts as the rest use states so there would be nothing else to change. Or you mean changing any `Statevector(QuantumCircuit)` to `Statevector(np.array)` for example?

**Comment by HuangJunye** [comment]
> > @Guillermo-Mijares-Vilarino Can you please simplify all the examples to use counts dictionary instead of result.get_counts similar to the second example based on Luciano's comment? [#8351 (comment)](https://github.com/Qiskit/qiskit-terra/pull/8351#issuecomment-1206306744)
> 
> If I do that at least to the first example from this PR it would become practically identical to one of the examples you added in #8569 @HuangJunye. We should then decide which one to keep.
> 
> This is the only of my code examples PRs that uses any counts as the rest use states so there would be nothing else to change. Or you mean changing any `Statevector(QuantumCircuit)` to `Statevector(np.array)` for example?

It's ok to have identical examples because the entry point of users could be from search engine not from the module api page. Yeah I think changing this one is enough. YOu can keep the statevector examples.

**Comment by Guillermo-Mijares-Vilarino** [comment]
Is there any other change that needs to be made @1ucian0?

**Review by 1ucian0 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**